RecognizeDtmf/RecognizeDtmf.cs

```diff
@@ -0,0 +1,334 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Calling.RecognizeDTMF
+{
+    using Azure.Communication;
+    using Azure.Communication.CallAutomation;
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    public class RecognizeDtmf
+    {
+        private CallConfiguration callConfiguration;
+        private CallAutomationClient callClient;
+        private CallConnection callConnection;
+        private CancellationTokenSource reportCancellationTokenSource;
+        private CancellationToken reportCancellationToken;
+
+        private TaskCompletionSource<bool> callEstablishedTask;
+        private TaskCompletionSource<bool> playAudioCompletedTask;
+        private TaskCompletionSource<bool> callTerminatedTask;
+        private TaskCompletionSource<bool> recognizeDtmfCompletedTask;
+        private DtmfTone toneInputValue = DtmfTone.Zero;
+        private int toneCount = 0;
+
+        public RecognizeDtmf(CallConfiguration callConfiguration)
+        {
+            this.callConfiguration = callConfiguration;
+            callClient = new CallAutomationClient(this.callConfiguration.ConnectionString);
+        }
+
+        public async Task Report(string targetPhoneNumber)
+        {
+            reportCancellationTokenSource = new CancellationTokenSource();
+            reportCancellationToken = reportCancellationTokenSource.Token;
+
+            try
+            {
+                callConnection = await CreateCallAsync(targetPhoneNumber).ConfigureAwait(false);
+                RegisterToDtmfResultEvent(callConnection.CallConnectionId);
+
+                await StartRecognizingDtmf(targetPhoneNumber).ConfigureAwait(false);
+                var recognizeDtmfCompleted = await recognizeDtmfCompletedTask.Task.ConfigureAwait(false);
+
+                if (recognizeDtmfCompleted)
+                {
+                    // dtmf tone found, let's play message based on key press.
+                    Logger.LogMessage(Logger.MessageType.INFORMATION, $"Play Audio for input {toneInputValue.ToString()}");
```

Comment:
Play Audio for invalid tone

Reply:
this input value change based on selection. Its not always 0.